### ISSUE TITLE
Feature Request: Ticket attachments

### ISSUE THREAD

**Issue by wrongecho** [itflow-org/itflow#739]
Allow attachments for tickets. Attachments would be uploaded and recorded in a new table. They would be accessible under the ticket reply associated with them (or under the main ticket if uploaded during ticket creation).

Table: **ticket_attachments**

ticket_attachment_id
ticket_attachment_name   (friendly name) 
ticket_attachment_ticket_id
ticket_attachment_ticket_reply_id   (0 is attached to the main ticket?)
ticket_attachment_ref (md5 / random name) 
ticket_attachment_created_at

**Comment by wrongecho** [comment]
This was completed under a3d8f5ab48e8332554ccdd06062d6c02a81da891

**Comment by boppy** [comment]
Reviewing the GUI and this commit, I only see the possibility to add attachments in the customer portal and by email. But I cannot add attachments as an admin in the "default" system view, or am I missing something (most likely I am... 🤷‍♂️)

**Comment by wrongecho** [comment]
> Reviewing the GUI and this commit, I only see the possibility to add attachments in the customer portal and by email. But I cannot add attachments as an admin in the "default" system view, or am I missing something (most likely I am... 🤷‍♂️)

You are completely correct. We need to add it on the agent side too! This should be relatively easy.

**Comment by wrongecho** [comment]
Merged in duplicate:

> Agents should be able to attach files as part of adding a ticket reply/comment.
For client-visible replies that send an email, either the file should either be attached to the email or a link to the file included at the bottom.

https://github.com/itflow-org/itflow/issues/1044#issue-2517911473

**Comment by wrongecho** [comment]
Moved to https://tasks.dev.itflow.org/task_details.php?task_id=104
